tests/test_proxy_info.py

```diff
@@ -13,107 +13,96 @@
+                    "format_version": 1,
+                    "upper_ota_proxy": "http://10.0.0.1:8082",
+                    "enable_local_ota_proxy": True,
+                    "enable_local_ota_proxy_cache": True,
+                    "local_ota_proxy_listen_addr": "0.0.0.0",
+                    "local_ota_proxy_listen_port": 8082,
+                    "logging_server": "http://10.0.0.1:8083",
+                }
+            ),
         ),
-        # case 3: testing invalid/corrupted proxy_info.yaml
+        # ------ case 3: invalid/corrupted proxy_info.yaml ------ #
         # If the proxy_info.yaml is not a yaml, otaclient will also treat
         # this case the same as proxy_info.yaml missing, the pre-defined
         # proxy_info.yaml will be used.
         (
             "not a valid proxy_info.yaml",
-            PARSED_PRE_DEFINED_PROXY_INFO_DICT,
+            DEFAULT_PROXY_INFO,
         ),
-        # case 4: testing default settings against invalid fields
-        # all config fields are expected to be assigned with default setting.
-        # NOTE: if proxy_info.yaml is valid yaml but fields are invalid,
-        #       default value settings will be applied.
-        # NOTE 2: not the same as [case1: proxy_info.yaml missing/corrupted]!
+        # ------ case 4: proxy_info.yaml is valid yaml but contains invalid fields ------ #
+        # in this case, default predefined default proxy_info.yaml will be loaded
+        # NOTE(20240126): Previous behavior is invalid field will be replaced by default value,
+        #                   and other fields will be preserved as much as possible.
+        #                 This is not proper, now the behavior changes to otaclient will treat
+        #                   the whole config file as invalid and load the default config.
         (
-            BAD_CONFIGURED_PROXY_INFO_YAML,
-            DEFAULT_SETTINGS_FOR_PROXY_INFO_DICT,
+            # Bad configured yaml file that contains invalid value.
+            # This yaml file is valid, but all fields' values are invalid.
+            (
+                "enable_local_ota_proxy: dafef\n"
+                "upper_ota_proxy: true\n"
+                "enable_local_ota_proxy_cache: adfaea\n"
+                "local_ota_proxy_listen_addr: 123\n"
+                'local_ota_proxy_listen_port: "2808"\n'
+            ),
+            DEFAULT_PROXY_INFO,
+        ),
+        # ------ case 5: corrupted/invalid yaml ------ #
+        # in this case, default predefined default proxy_info.yaml will be loaded
+        (
+            "/t/t/t/t/t/t/t/tyaml file should not contain tabs/t/t/t/",
+            DEFAULT_PROXY_INFO,
+        ),
+        # ------ case 6: backward compatibility test ------ #
+        # for superseded field, the corresponding field should be assigned.
+        # for removed field, it should not impact the config file loading.
+        (
+            "enable_ota_proxy: true\ngateway: false\n",
+            DEFAULT_PROXY_INFO,
```

Comment:
Test case 6 is added to ensure the backward-compatibility with old proxy_info.yaml.
1. `enable_ota_proxy` field's value will be assigned to new `enable_local_ota_proxy` field.
2. deprecated `gateway` field will be ignored when parsing the proxy_info.yaml.